### PR TITLE
feat: expose polyhash clustering algorithms under UnlCore.Polyhash

### PR DESCRIPTION
**Pull request by jake-unl**
@eturan 

Here is a clunky integration that exposes all polyhash related methods.
We are still deciding whether all of these are useful to 3rd party developers

TODO:

- refine exposed API
- remove reliance on 3rd party polygon utilities from Turf

### CODE REVIEW

**Comment by eturan** [comment]
@cristian-calugar could you please take a look as well?

File: polyhash/README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+### decompressPolyhash(compressedPolyhash)
+
+Decompress a polyhash.
+
+```js
+const compressedPolyhash = 'ZNYZN3Y1xNyA';
+const polyhash = decompressPolyhash(compressedPolyhash);
+
+/*
+  polyhash = ['tester','sp','j','ek']
```
Comment (cristian-calugar) [inline polyhash/README.md:68]: I think that here is missing the last item from polihash: 'er'

File: polyhash/README.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+]
+*/
+```
+
+### inflate(polyhash)
+
+Convert a polyhash to full locationIds
+
+```js
+     [{ precision: 4, data: [ 'drsv' ] },
```
Comment (cristian-calugar) [inline polyhash/README.md:96]: It is not clear why this example contains the lines 96 to 103

File: polyhash/README.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+*/
+```
+
+### deflate(locationIds)
+
+Convert an array of locationIds to polyhash, used internally for binary compression
+
+```js
+const locationIds = ['drsv', 'drtjb', 'drtj8', 'drtj2', 'drtj0'];
+const polyhash = deflate(polyhash);
```
Comment (cristian-calugar) [inline polyhash/README.md:126]: I think this should be: const polyhash = deflate(locationIds);

File: polyhash/README.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+*/
+```
+
+### groupByPrefix(locationIds)
+
+Convert an array of locationIds to polyhash
+
+```js
+const locationIds = ['drsv', 'drtjb', 'drtj8', 'drtj2', 'drtj0'];
+const polyhash = groupByPrefix(polyhash);
```
Comment (cristian-calugar) [inline polyhash/README.md:142]: same here, wrong parameter

File: polyhash/index.js [hunk 5]
```diff
@@ -0,0 +1,459 @@
+const unl = require('../unl-core');
+const turfBooleanContains = require('@turf/boolean-contains').default;
+const turfBooleanDisjoint = require('@turf/boolean-disjoint').default;
+const turfIntersect = require('@turf/intersect').default;
+const turfHelpers = require('@turf/helpers')
+const turfMeta = require('@turf/meta')
+
+const maxPrecision = 16;
+
+var Clustering = {};
```
Comment (cristian-calugar) [inline polyhash/index.js:10]: We should add also some tests for polyhash methods

**Review by cristian-calugar — APPROVED** [review]

**Review by eturan — APPROVED** [review]

**Pull request merged**